# https://github.com/qtile/qtile/pull/5074

--- site-packages/libqtile/backend/base/drawer.py
+++ site-packages/libqtile/backend/base/drawer.py
@@ -312,6 +312,8 @@
 
     def clear(self, colour):
         """Clears background of the Drawer and fills with specified colour."""
+        if self.ctx is None:
+            self._reset_surface()
         self.ctx.save()
 
         # Erase the background
